test/utils.go

```diff
@@ -0,0 +1,133 @@
+package test
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+)
+
+type Request struct {
+	Method             string
+	Uri                string
+	ResponseStatusCode int
+	ResponseBody       interface{}
+}
+
+type server struct {
+	URL                string // Shortcut to httptest.Server.URL
+	server             *httptest.Server
+	requestCounter     int
+	nbExpectedRequests int
+	t                  *testing.T
+}
+
+// MockHttpServer creates a test server which will send responses in the given order
+// It is possible to check on Method and Uri if set
+// Responses can contain JSON-encoded body if set
+func MockHttpServer(t *testing.T, requests []Request) *server {
+	s := server{
+		requestCounter:     0,
+		nbExpectedRequests: len(requests),
+		t:                  t,
+	}
+
+	s.server = httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var response Request
+		response, requests = requests[0], requests[1:]
```

Comment:
`requests` set but not used?

Reply:
Oh, it will contain requests for the future calls to the server... This is very cool 💯

Reply:
Thank for the comment, it helped me see that I forgot to rename response to request (for the current request (singular))